packages/excubiae/contracts/extensions/ZKEdDSAEventTicketPCDExcubia.sol

```diff
@@ -0,0 +1,96 @@
+/// @dev The contract uses specific event ID and signers to check against the verifier
+/// in order to admit the recipient (passerby) of the proof.
+contract ZKEdDSAEventTicketPCDExcubia is Excubia {
+    /// @notice The valid event ID that proofs must match to pass the gate.
+    uint256 public immutable VALID_EVENT_ID;
+    /// @notice The first valid signer whose signatures are considered valid to pass the gate.
+    uint256 public immutable VALID_SIGNER_1;
+    /// @notice The second valid signer whose signatures are considered valid to pass the gate.
+    uint256 public immutable VALID_SIGNER_2;
+
+    /// @notice The ZKEdDSA Event Ticket PCD Verifier contract.
+    ZKEdDSAEventTicketPCDVerifier public immutable VERIFIER;
+
+    /// @notice Mapping to track which tickets have already passed the checks
+    /// to avoid passing the gate twice with the same ticket.
+    mapping(uint256 => bool) public passedZKEdDSAEventTicketPCDs;
+
+    /// @notice Error thrown when the proof is invalid.
+    error InvalidProof();
+
+    /// @notice Error thrown when the event ID in the proof does not match the valid event ID.
+    error InvalidEventId();
+
+    /// @notice Error thrown when the signers in the proof do not match the valid signers.
+    error InvalidSigners();
+
+    /// @notice Error thrown when the watermark in the proof does not match the passerby address.
+    error InvalidWatermark();
+
+    /// @notice Constructor to initialize with target verifier, valid event ID, and valid signers.
+    /// @param _verifier The address of the ZKEdDSA Event Ticket PCD Verifier contract.
+    /// @param _validEventId The valid event ID that proofs must match.
+    /// @param _validSigner1 The first valid signer whose signatures are considered valid.
+    /// @param _validSigner2 The second valid signer whose signatures are considered valid.
+    constructor(address _verifier, uint256 _validEventId, uint256 _validSigner1, uint256 _validSigner2) {
+        if (_verifier == address(0)) revert ZeroAddress();
+
+        VERIFIER = ZKEdDSAEventTicketPCDVerifier(_verifier);
+        VALID_EVENT_ID = _validEventId;
+        VALID_SIGNER_1 = _validSigner1;
+        VALID_SIGNER_2 = _validSigner2;
+    }
+
+    /// @notice Internal function to handle the passing logic with check.
+    /// @dev Calls the parent `_pass` function and stores the ticket ID to avoid passing the gate twice.
+    /// @param passerby The address of the entity attempting to pass the gate.
+    /// @param data Additional data required for the check (i.e., encoded proof).
+    function _pass(address passerby, bytes calldata data) internal override {
+        // Decode the given data bytes.
+        (, , , uint256[38] memory _pubSignals) = abi.decode(data, (uint256[2], uint256[2][2], uint256[2], uint256[38]));
+
+        // Ticket ID is stored at index 0.
+        uint256 ticketId = _pubSignals[0];
```

Comment:
is it more efficient to create a new var rather than using _pubSignals[0] twice down in the next lines?